spring-integration-http/src/main/java/org/springframework/integration/http/outbound/HttpRequestExecutingMessageHandler.java

```diff
@@ -276,354 +135,35 @@ public void setErrorHandler(ResponseErrorHandler errorHandler) {
 	 *
 	 * @see RestTemplate#setMessageConverters(java.util.List)
 	 */
+	@Override
 	public void setMessageConverters(List<HttpMessageConverter<?>> messageConverters) {
 		this.restTemplate.setMessageConverters(messageConverters);
 	}
 
 	/**
-	 * Set the {@link HeaderMapper} to use when mapping between HTTP headers and MessageHeaders.
-	 * @param headerMapper The header mapper.
-	 */
-	public void setHeaderMapper(HeaderMapper<HttpHeaders> headerMapper) {
-		Assert.notNull(headerMapper, "headerMapper must not be null");
-		this.headerMapper = headerMapper;
-	}
-
-	/**
 	 * Set the {@link ClientHttpRequestFactory} for the underlying {@link RestTemplate}.
 	 *
 	 * @param requestFactory The request factory.
 	 *
 	 * @see RestTemplate#setRequestFactory(ClientHttpRequestFactory)
 	 */
 	public void setRequestFactory(ClientHttpRequestFactory requestFactory) {
-		this.restTemplate.setRequestFactory(requestFactory);
-	}
-
-	/**
-	 * Set the Map of URI variable expressions to evaluate against the outbound message
-	 * when replacing the variable placeholders in a URI template.
-	 *
-	 * @param uriVariableExpressions The URI variable expressions.
-	 */
-	public void setUriVariableExpressions(Map<String, Expression> uriVariableExpressions) {
-		synchronized (this.uriVariableExpressions) {
-			this.uriVariableExpressions.clear();
-			this.uriVariableExpressions.putAll(uriVariableExpressions);
+		if (this.isAsync()) {
+			throw new IllegalArgumentException("ClientHttpRequestFactory cannot be set in async mode");
```

Comment:
I think we shouldn't be so strict.
The `async` mode will be ignored anyway, because this component doesn't return `ListenableFuture`.
See `AbstractMessageProducingHandler`:
```
if (this.async && reply instanceof ListenableFuture<?>) {
			ListenableFuture<?> future = (ListenableFuture<?>) reply;
```
So, we should just silently close our eyes for what user has configured for the `HttpRequestExecutingMessageHandler.async`